android/app/src/main/java/org/fossasia/openevent/fragments/SpeakersListFragment.java

```diff
@@ -77,7 +78,12 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
 
         View view = super.onCreateView(inflater, container, savedInstanceState);
 
-        OpenEventApp.getEventBus().register(this);
+        if (Utils.isBaseUrlEmpty()) {
+            swipeRefreshLayout.setEnabled(false);
+        } else {
+            OpenEventApp.getEventBus().register(this);
+            swipeRefreshLayout.setOnRefreshListener(this::refresh);
+        }
```

Comment:
I was talking about this whole block being repeating. Create a static method encapsulating the whole registering unregistering flow.

Likde `registerIfUrlValid`
Arguments can be `swipeRefereshLayout`, `this` and `this::refresh` for each fragment

And similarly `unregisterIfUrlValid`, taking only `this`